ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilder.java

```diff
@@ -318,6 +345,60 @@ private static PhysicalSchema getPhysicalSchema(final SourceStep<?> streamSource
         .transformValues(new AddKeyAndTimestampColumns<>(keyGenerator));
   }
 
+  private static void registerChangelogSourceTopic(
+      final String name,
+      final KsqlQueryBuilder queryBuilder,
+      final String sourceSubject
+  ) {
+    final String changelogTopic = changelogTopic(queryBuilder, name);
+    final SchemaRegistryClient srClient = queryBuilder
+        .getServiceContext()
+        .getSchemaRegistryClient();
+    final KafkaTopicClient topicClient = queryBuilder.getServiceContext().getTopicClient();
+
+    try {
+      final Collection<String> allSubjects = srClient.getAllSubjects();
+      // if the source subject isn't using schema registry, don't register a
+      // schema; also if this topic actually exists, we shouldn't add the "fake"
+      // schema to it
+      if (!allSubjects.contains(sourceSubject) || topicClient.isTopicExists(changelogTopic)) {
+        return;
+      }
+
+      final List<Integer> allVersions = srClient.getAllVersions(sourceSubject);
+      for (final Integer version : allVersions) {
```

Comment:
How does registering an existing schema under a different subject contribute to Schema Registry usage (e.g., in CCloud)? Are we concerned about use cases where a user has many versions of the source subject and ksqlDB registers many new schemas?

Reply:
Yes, unfortunately it does count twice against the schema registry - but I think this is acceptable because all future queries will anyway have this because the changelog topic will be there.